File: Two-Pointers/125-Valid-Palindrome.py
def isPalindrome(s):
    left = 0
    right = len(s)-1
    while left < right:
        # Step 1
        while left < right and not s[left].isalnum():
            left += 1
        #Step 2
        while left < right and not s[right].isalnum():
            right -= 1
        # Steps 3 and 4
        if s[left].lower() != s[right].lower():
            return False
        else:
            left += 1
            right -= 1
    # Step 6
    return True

print(isPalindrome("A man, a plan, a canal: Panama"))

File: Two-Pointers/125-Valid-Palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */

 //1. we want to remove all non-alphanumeric characters from string
 //2. turn all letters into lowercase
 //3. iterate through string with two pointers, x and y
 //3.5 declare two points within an array, x and y
 //4. if x and y are the same after iterating half of the word, return true
 //5. otherwise, return false

 var isPalindrome = function(s) {
  let cleanUpArr = s.split(' ').join('');
  const getLettersOnly = /[\W_]/g;
  let finalArr = cleanUpArr.replace(getLettersOnly, '').toLowerCase();
  let y = finalArr.length;

  for (let x=0; x<finalArr.length/2; x++) {
      if (finalArr[x] !== finalArr[y - 1 - x]) {
         return false;
      };
  };
  return true;
};

console.log(isPalindrome("A man, a plan, a canal: Panama"));